katello/katello-restore

```diff
@@ -16,72 +16,88 @@ optparse = OptionParser.new do |opts|
   opts.on("-y", "--assumeyes", "Answer yes for all questions") do
     confirmed = true
   end
+
+  opts.parse!
+
+  if ARGV.length == 0
+    opts.abort("**** ERROR: Please specify the backup directory to restore ****")
+  elsif ARGV.length != 1
+    puts opts
+    exit(-1)
+  end
+
+  @dir = ARGV.pop.dup
+  unless File.directory?(@dir)
+    opts.abort("Backup directory does not exist: #{@dir}")
+  end
 end
 
-optparse.parse!
-DIR = ARGV.pop.dup
-DIR.chop! if DIR.end_with?("/")
+def run_cmd(command, exit_codes=[0])
+  `#{command}`
+  unless exit_codes.include?($?.exitstatus)
+    STDERR.puts "Failed '#{command}'"
+    exit(-1)
+  end
+end
 
 def set_file_security
   puts "Setting file security"
-  `restorecon -Rnv /`
+  run_cmd "restorecon -Rnv / 2>/dev/null"
   puts "Done.\n"
 end
 
 def reset_katello
   puts "Resetting Katello"
-  `tar --selinux --overwrite --listed-incremental=/dev/null -xzf config_files.tar.gz -C /`
-  `foreman-installer --scenario katello --reset`
+  run_cmd "tar --selinux --overwrite --listed-incremental=/dev/null -xzf config_files.tar.gz -C /"
+  run_cmd("foreman-installer --scenario katello --reset", [0,6])
```

Comment:
@cfouant By latest snaps do you mean 3.4? 

Since this command is currently broken in master, my thought is to do this to avoid any issues:
- use "yes" workaround in this PR
- fix in foreman-installer with -y flag or similar
- remove "yes" workaround once foreman-installer flag can be used